documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -52,178 +48,164 @@ public void run(String routingKey, String message) {
+
+3. **Event-based:** Similar to the data-based scaling policy, but it uses number of Events instead of bytes.
+
+The minimum number of Segments is a factor that sets the minimum degree of read parallelism to be maintained; for example if this value is set as three, there will always be three Stream Segments available on the Stream.  Currently, this property is effective only when the stream is
 created; at some point in the future, update stream will allow this factor to be
 used to change the minimum degree of read parallelism on an existing Stream.
 
-Once the `StreamConfiguration` object is created, creating the Stream is straight
-forward (line 8).  After the Stream is created, we are all set to start writing
+Once the Stream Configuration (`streamConfig`) object is created, creating the Stream is straight
+forward using `createStream()`. After the Stream is created, we are all set to start writing
 Event(s) to the Stream.
 
+
 ## Writing an Event using EventWriter
 
 Applications use an `EventStreamWriter` object to write Events to a Stream.  The
-key object to creating the `EventStreamWriter` is the ClientFactory.  The
-ClientFactory is used to create Readers, Writers and other types of Pravega
+key object to creating the `EventStreamWriter` is the `ClientFactory`.  The
+`ClientFactory` is used to create Readers, Writers and other types of Pravega
 Client objects such as the State Synchronizer (see [Working with Pravega: State
 Synchronizer](state-synchronizer.md)).
 
-Line 10 shows the creation of a ClientFactory.  A ClientFactory is created in
-the context of a Scope, since all Readers, Writers and other Clients created by
-the ClientFactory are created in the context of that Scope.  The ClientFactory
-also needs a URI to one of the Pravega Controllers, just like StreamManager.
+A `ClientFactory` is created in the context of a Scope, since all Readers, Writers and other Clients created by the `ClientFactory` are created in the context of that Scope. The `ClientFactory`
+also needs a URI to one of the Pravega Controllers (`ClientFactory.withScope(scope, controllerURI)`) , just like `StreamManager`.
 
-Because ClientFactory and the objects it creates consumes resources from
+As the `ClientFactory` and the objects it creates consumes resources from
 Pravega, it is a good practice to create these objects in a try-with-resources
```

Comment:
@shrids Try-with-resources...( i am not clear about the meaning conveyed, could u rephrase??)

Reply:
@shirds Thanks for ref....